src/box/box.cc

```diff
@@ -4436,6 +4461,37 @@ box_cfg(void)
 	}
 }
 
+static const char *recovery_state_names[RECOVERY_STATE_MAX] = {
+	/* [RECOVERY_STATE_SNAPSHOT_RECOVERED] = */
```

Comment:
Nit: You don't need to comment it. In modern C you could just as well write

```C
[RECOVERY_STATE_SNAPSHOT_RECOVERED] = "snapshot_recovered",
```

and we quite often do that.

Reply:
Yes, and I did that initially, but this file is C++, and some compilers error with "designated array initializers are a C extension" or something like that. It was caught with full-ci, AFAIR.

BTW, C++20 allows designated initializers for structs, but not for arrays.

It would look nicer this way, indeed.

Reply:
Oh, I missed that it's C++. Okay then.